code/modules/projectiles/guns/chaos_bolt.dm

```diff
@@ -0,0 +1,348 @@
+			if(target_heart)
+				target_heart.remove(H)
+				qdel(target_heart)
+		if("gibbed")
+			H.visible_message("<span class='chaosverybad'>[H] falls into gibs!</span>", "<span class='chaosverybad'>Oof!</span>")
+			H.gib()
+		if("cluwned")
+			H.visible_message("<span class='chaosverybad'>[H] turns into a cluwne!</span>", "<span class='chaosverybad'>Oh no.</span>")
+			H.makeCluwne()
+		if("spaced")
+			for(var/obj/item/I in H)
+				H.unEquip(I, TRUE)
+			var/turf/T = safepick(get_area_turfs(/area/space/nearstation)) //Send in space next to the station
+			if(!T) //Shouldn't happen but just in case
+				T = safepick(get_area_turfs(/area/space))
+			if(!T) //What do you mean there's no space? Okay well just die then
+				H.visible_message("<span class='chaosverybad'>[H] drops dead!</span>", "<span class='chaosverybad'>Game over!</span>")
+				H.death(FALSE)
+			else
+				H.visible_message("<span class='chaosverybad'>[H] disappears!</span>", "<span class='chaosverybad'>COLD! CAN'T BREATH!</span>")
+				do_teleport(H, T)
+		if("decapitated")
+			H.visible_message("<span class='chaosverybad'>[H]'s head goes flying!'</span>", \
+				"<span class='chaosverybad'>You watch the floor fly to your face as you rapidly lose consciousness...</span>")
+			var/obj/item/organ/external/affected = target.get_organ("head")
+			var/atom/movable/A = affected.droplimb(1, DROPLIMB_SHARP)
+			INVOKE_ASYNC(A, TYPE_PROC_REF(/atom/movable, throw_at), pick(oview(7, get_turf(src))), 10, 1)
+		if("banned")
+			H.visible_message("<span class='chaosverybad'>[H] gets <span class='adminhelp'>BWOINKED</span> out of existence!</span>", \
+				"<span class='chaosverybad'>You get <span class='adminhelp'>BWOINKED</span> out of existence!</span>")
+			playsound(H, 'sound/effects/adminhelp.ogg', 100, FALSE)
+			qdel(H)
+		if("exploded")
+			H.visible_message("<span class='chaosverybad'>[H] explodes!</span>", "<span class='chaosverybad'>Boom!</span>")
+			explosion(get_turf(H), 1, 1, 1, cause = "staff of chaos lethal explosion effect")
+		if("cheese morphed")
+			H.visible_message("<span class='chaosverybad'>[H] transforms into cheese!</span>", "<span class='chaosverybad'>You've been transformed into cheese!</span>")
+			new /obj/item/reagent_containers/food/snacks/cheesewedge(get_turf(H))
+			qdel(H)
+		if("supermattered")
+			var/obj/machinery/atmospherics/supermatter_crystal/supercrystal = locate(/obj/machinery/atmospherics/supermatter_crystal)
+			if(!supercrystal)
+				H.visible_message("<span class='chaosverybad'>[H] drops dead!</span>", "<span class='chaosverybad'>Game over!</span>")
+				H.death()
+			else
+				H.visible_message("<span class='chaosverybad'>[H] disappears!</span>", "<span class='chaosverybad'>All you see is yellow before you fall to dust...</span>")
+				do_teleport(H, supercrystal, 1)
+				H.throw_at(supercrystal, 10, 2)
+				if(H && H.stat == CONSCIOUS)
+					to_chat(H, "<span class='chaosverybad'>... not? You're alive? Huh. Neat.</span>")
```

Comment:
this references supercrystal even though it's known to be null here

Reply:
This is part of the else clause where supermatter is not null

Reply:
oop you're right